specs/~dynamic-names.yml

```diff
@@ -0,0 +1,212 @@
+overview: |
+  Dynamic Names are a special notation to dynamically determine a tag's content.
+
+  Dynamic Names MUST be a non-whitespace character sequence NOT containing
+  the current closing delimiter. A Dynamic Name consists of an asterisk,
+  followed by a dotted name. The latter follows the same notation as in an
+  Interpolation tag.
+
+  This tag's dotted name, which is equal to the Dynamic Name excluding the
+  leading asterisk, refers to a key in the context whose value will be used in
+  place of the Dynamic Name itself as content of the tag. The dotted name
+  resolution produces the same value as an Interpolation tag and does not affect
+  the context for further processing.
+
+  This tag's content refers to a key in the context whose value will be used in
+  place of the Dynamic Name itself as content of the tag. The name resolution is
+  identical to name resolution in Interpolation tags, as follows:
+    1) Split the name on periods; the first part is the name to resolve, any
+    remaining parts should be retained.
+    2) Walk the context stack from top to bottom, finding the first context
+    that is a) a hash containing the name as a key OR b) an object responding
+    to a method with the given name.
+    3) If the context is a hash, the data is the value associated with the
+    name.
+    4) If the context is an object, the data is the value returned by the
+    method with the given name.
+    5) If any name parts were retained in step 1, each should be resolved
+    against a context stack containing only the result from the former
+    resolution.  If any part fails resolution, the result should be considered
+    falsey, and should interpolate as the empty string.
+  The resolved data should be coerced into a string before being used as
+  content.
```

Comment:
I think this paragraph can go now, as it is mostly redundant with the previous paragraph.